src/source/nats.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![cfg(not(tarpaulin_include))]
+
+use crate::sink::nats::ConnectOptions;
+use crate::source::prelude::*;
+use async_nats::{Connection as NatsConnection, Subscription};
+
+#[derive(Debug, Clone, Deserialize, Default)]
+pub struct Config {
+    // list of hosts
+    pub hosts: Vec<String>,
+    // subject to send messages to
+    pub subject: String,
+    // optional queue to subscribe to
+    #[serde(default = "Default::default")]
+    pub queue: Option<String>,
+    // options to use when opening a new connection
+    #[serde(default = "Default::default")]
+    pub options: ConnectOptions,
+}
+
+impl ConfigImpl for Config {}
+
+impl Config {
+    async fn connection(&self) -> Result<NatsConnection> {
+        let hosts = self.hosts.join(",");
+        let connection = self.options.generate()?.connect(hosts.as_str()).await?;
```

Comment:
Where a `&str` is required and you have a `String` most of the time just using `&` works fine and is the more rusty way :D 
```suggestion
        let connection = self.options.generate()?.connect(&hosts).await?;
```